### DISCUSSION TITLE
Upload PDF Component to Azure Blob

### DISCUSSION THREAD

**Discussion by viswanadula8828** [reflex-dev/reflex#2513] [Q&A] [answered]
How to display uploaded files on the UI using reflex libraries. I tried, but didn't find easier way. 

I am trying below things. I was able to acheive with text box, but was not sucessful with pdf documents

![image](https://github.com/reflex-dev/reflex/assets/144563900/6800e80e-d532-422d-9b66-0ae5729a7784)

1. Upload pdf to a blob storage in Azure
2. Display on UI stating that pdf it is uploaded sucessfully
3. Summarize the pdf through the use of langchain
4. Display on the screen the summarized version for the users

Need help with Point 1 and Point 2.!!

**Comment by masenf** [comment]
Couple of clarifying questions:

1. does the PDF need to be uploaded to Azure?
2. Is the langchain instance running on the same host as the reflex app, or accessed via API?

**Reply by viswanadula8828** [reply]
- Not necessarily. I tried uploading the pdf to the .web folder and added os.remove to delete those files after the summarization. Let's say, if I would like to know the logs of who and what they uploaded and deleted, how can I get them?
- how can I get general user stats like how many people got into the application?
- Yes, langchain instance is accessed by API

**Reply by masenf** [reply, marked as answer]
have you played with the `rx.upload` component as described here https://reflex.dev/docs/library/forms/upload/?

You can get the raw bytes of the file in the event handler and then send that off to the API directly if it accepts bytes. If it only accepts a URL, then i guess you will have to upload it somewhere.

From the upload event handler though you can get access to the user information recorded in the state (assuming that you're tracking users already in some way)

**Reply by viswanadula8828** [reply]
Worked out with upload docs. Thank you
